lib/fennec/turn.ex

```diff
@@ -2,13 +2,19 @@ defmodule Fennec.TURN do
   @moduledoc false
   # This module defines a struct used as TURN protocol state.
 
-  defstruct allocation: nil, permissions: %{}, channels: [],
-            nonce: nil, realm: nil
+  alias Jerboa.Format
+  alias Fennec.TURN.Channel
+
+  defstruct allocation: nil, permissions: %{}, channels: {%{}, %{}},
+    nonce: nil, realm: nil
+
+  @permission_lifetime 5 * 60 # MUST be 5mins
 
   @type t :: %__MODULE__{
     allocation: nil | Fennec.TURN.Allocation.t,
     permissions: %{peer_addr :: Fennec.ip => expiration_time :: integer},
-    channels: [],
+    channels: {%{peer :: Fennec.address => Channel.t},
+               %{Format.channel_number  => Channel.t}},
```

Comment:
If I remember correctly there was tricky logic around these in Icicle so perhaps simple is the way to go (to begin with, like you say, at least).